src/staticanalysis/bot/static_analysis_bot/coverity/__init__.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import io
+import os
+import requests
+import subprocess
+import tarfile
+import cli_common.log
+import cli_common.utils
+from static_analysis_bot.config import settings
+from static_analysis_bot import AnalysisException
+
+
+logger = cli_common.log.get_logger(__name__)
+
+COVERITY_CONFIG = '''
+  {
+    "type": "Coverity configuration",
+    "format_version": 1,
+    "settings": {
+      "server": {
+        "host": "%s",
+        "ssl" : true,
+        "on_new_cert" : "trust",
+        "auth_key_file": "%s"
+      },
+      "stream": "Firefox",
+      "cov_run_desktop": {
+        "build_cmd": [],
+        "clean_cmd": []
+      }
+    }
+  }
+'''
+
+
+def setup(index):
+    '''
+    Setup Taskcluster Coverity build for static-analysis
+    '''
+    assert settings.cov_url, 'Missing secret COVERITY_CONFIG:server_url'
+    assert settings.cov_analysis_url, 'Missing secret COVERITY_CONFIG:package_url'
+    assert settings.cov_auth, 'Missing secret COVERITY_CONFIG:auth_key'
+
+    target = os.path.join(
+        os.environ['MOZBUILD_STATE_PATH'], 'coverity')
+
+    # Generate the coverity.conf and auth files
+    cov_auth_path = os.path.join(target, 'auth')
+    cov_setup_path = os.path.join(target, 'coverity.conf')
+    cov_conf = COVERITY_CONFIG % (settings.cov_url, cov_auth_path)
+
+    logger.info('Downloading from {}.'.format(settings.cov_analysis_url))
+    cli_common.utils.retry(lambda: download(settings.cov_analysis_url, target))
+    if not os.path.exists(target):
+        raise AnalysisException('artifact', 'Setup failed for {}'.format(target))
+
+    with open(cov_auth_path, 'a') as f:
+        f.write(settings.cov_auth)
+
+    # Modify it's permission to 600
+    subprocess.run(['chmod', '600', cov_auth_path])
```

Comment:
I did this because I've also seen accepted [here](https://github.com/abpostelnicu/release-services/blob/master/src%2Fstaticanalysis%2Fbot%2Fstatic_analysis_bot%2Finfer%2F__init__.py#L107)